### DISCUSSION TITLE
Unsat core when sygus returns infeasible

### DISCUSSION THREAD

**Discussion by zhengyao-lin** [cvc5/cvc5#10806] [Q&A] [answered]
Hello! When sygus returns infeasible to a synthesis problem, is it possible to get the reason of the infeasibility somehow?
For example, if there is a constraint which makes the problem infeasible, is it possible to get it like an unsat core?

Thank you!

**Comment by ajreynol** [comment, marked as answer]
We don't currently have support for this, but this feature is on our radar.

For some background, SyGuS solving is quite a bit different from SMT solving so the same machinery does not apply.  In particular, there are several reasons why the SyGuS solver answers "infeasible", including:
(1) It encountered a "CEGIS refinement lemma" that was equivalent to false,
(2) It consulted a decision procedure for the synthesis problem which did not find a solution,
(3) It exhausted all possible terms in a finite grammar.

The interfaces for each of these reasons would all be different. Nevertheless we would like to provide this information in some form, we will keep you posted.
